<launch>
    <arg name="raw" default="false" doc="If set to true, also publish raw image using decoder."/>
    <arg name="cam_info" default="true" doc="If set to true, use cam_info_reader_node to pusblish CameraInfo."/>
    <arg name="live" default="false" doc="live set to false means we don't run the actual camera (i.e. running from a log file" />
    <group if="$(arg live)">
        <node name="camera_node" pkg="pi_camera" type="camera_node.py" output="screen" required="true"/>
    </group>

    <group if="$(arg raw)">
        <remap from="decoder_node/compressed_image" to="camera_node/image/compressed"/>
        <remap from="decoder_node/image/raw" to="camera_node/image/raw"/>
        <node name="decoder_node" pkg="pi_camera" type="decoder_node.py" output="screen" required="true"/>
    </group>

    <group if="$(arg cam_info)">
        <remap from="cam_info_reader_node/camera_info" to="camera_node/camera_info"/>
        <remap from="cam_info_reader_node/compressed_image" to="camera_node/image/compressed"/>
        <node name="cam_info_reader_node" pkg="pi_camera" type="cam_info_reader_node.py" output="screen" required="true"/>
           <param name="~image_type" value="compressed" />

        <group if="$(arg raw)">
            <remap from="raw_cam_info_reader_node/camera_info" to="camera_node/raw_camera_info"/>
            <remap from="raw_cam_info_reader_node/compressed_image" to="camera_node/image/raw"/>
            <node name="raw_cam_info_reader_node" pkg="pi_camera" type="cam_info_reader_node.py" output="screen" required="true"/>
                <param name="~image_type" value="raw" />
        </group>
    </group>


</launch>
